jax/interpreters/xla.py

```diff
@@ -533,6 +602,52 @@ def _xla_callable(fun: lu.WrappedFun, device, backend, name, *arg_specs):
   else:
     return partial(_execute_replicated, compiled, result_handlers)
 
+def configure_aliasing(
+    c: xc.XlaBuilder,
+    xla_args: Sequence[xe.XlaOp],
+    out_tuple: xe.XlaOp,
+    arg_donate: Sequence[bool],
+    tuple_args: bool,
+):
+  """Configures input/output "must" aliasing based on `arg_donate`."""
+  assert len(xla_args) == len(arg_donate)
+
+  # First for every input array add it to `donations` iff it is a member of
+  # `arg_donate`.
+  donations = defaultdict(deque)
+  for arg_index, arg in enumerate(xla_args):
```

Comment:
This code is complex enough that it deserves unit tests?

Reply:
We test the result of this (e.g. on supported platforms input buffers are deleted and output, if there is no match we raise a warning) but I haven't unit tested the method directly, it feels like an implementation detail to me. Do you feel strongly that this is tested in isolation? To do this well I would need to refactor it a bit to not require us to build up a computation.

Reply:
I'm going to assume the current tests are fine and close this out, but please re-open if you feel strongly.